provider/datagen/src/transform/cldr/relativetime/mod.rs

```diff
@@ -0,0 +1,219 @@
+            ),
+            (LongQuarterRelativeTimeFormatDataV1Marker::KEY, "quarter"),
+            (
+                ShortQuarterRelativeTimeFormatDataV1Marker::KEY,
+                "quarter-short",
+            ),
+            (
+                NarrowQuarterRelativeTimeFormatDataV1Marker::KEY,
+                "quarter-narrow",
+            ),
+            (LongYearRelativeTimeFormatDataV1Marker::KEY, "year"),
+            (ShortYearRelativeTimeFormatDataV1Marker::KEY, "year-short"),
+            (NarrowYearRelativeTimeFormatDataV1Marker::KEY, "year-narrow"),
+        ]
+        .into_iter()
+        .collect()
+    };
+}
+
+macro_rules! make_data_provider {
+    ($($marker: ident),+ $(,)?) => {
+        $(
+            impl DataProvider<$marker> for crate::DatagenProvider {
+                fn load(
+                    &self,
+                    req: DataRequest,
+                ) -> Result<DataResponse<$marker>, DataError> {
+                    let langid = req.locale.get_langid();
+                    let resource: &cldr_serde::date_fields::Resource = self
+                        .source
+                        .cldr()?
+                        .date_fields()
+                        .read_and_parse(&langid, "dateFields.json")?;
+                    let fields = &resource
+                        .main
+                        .0
+                        .get(&langid)
+                        .ok_or(DataErrorKind::MissingLocale.into_error())?
+                        .dates
+                        .fields;
+
+                    let field = DATAKEY_FILTERS
+                        .get(&$marker::KEY)
+                        .ok_or(DataErrorKind::MissingDataKey.into_error())?;
+
+                    let data = fields.0.get(*field).ok_or(DataError::custom(
+                        "Field not found in relative time format data.",
+                    ))?;
+
+                    Ok(DataResponse {
+                        metadata: Default::default(),
+                        payload: Some(DataPayload::from_owned(data.try_into()?)),
+                    })
+                }
+            }
+
+            impl IterableDataProvider<$marker> for crate::DatagenProvider {
+                fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+                    Ok(self
+                        .source
+                        .cldr()?
+                        .date_fields()
+                        .list_langs()?
+                        .map(DataLocale::from)
+                        .collect())
+                }
+            }
+
+        )+
+    };
+}
+
+impl TryFrom<&cldr_serde::date_fields::Field> for RelativeTimePatternDataV1<'_> {
+    type Error = DataError;
+    fn try_from(field: &cldr_serde::date_fields::Field) -> Result<Self, Self::Error> {
+        let mut relatives = ZeroMap::new();
+        for relative in &field.relatives {
+            relatives.insert(&relative.count, relative.pattern.as_ref());
```

Comment:
note: filed https://github.com/unicode-org/icu4x/issues/2826 to get us to be more efficient around this

Reply:
Added a TODO here.